Changing manpage installation paths to where debian wants them.

Index: fdm-1.6/GNUmakefile
===================================================================
--- fdm-1.6.orig/GNUmakefile	2009-04-14 06:00:40.000000000 +0200
+++ fdm-1.6/GNUmakefile	2009-04-14 06:00:51.000000000 +0200
@@ -115,10 +115,10 @@
 install:
 	$(INSTALLDIR) $(DESTDIR)$(PREFIX)/bin
 	$(INSTALLBIN) $(PROG) $(DESTDIR)$(PREFIX)/bin/$(PROG)
-	$(INSTALLDIR) $(DESTDIR)$(PREFIX)/man/man1
-	$(INSTALLMAN) $(PROG).1 $(DESTDIR)$(PREFIX)/man/man1/$(PROG).1
-	$(INSTALLDIR) $(DESTDIR)$(PREFIX)/man/man5
-	$(INSTALLMAN) $(PROG).conf.5 $(DESTDIR)$(PREFIX)/man/man5/$(PROG).conf.5
+	$(INSTALLDIR) $(DESTDIR)$(PREFIX)/share/man/man1
+	$(INSTALLMAN) $(PROG).1 $(DESTDIR)$(PREFIX)/share/man/man1/$(PROG).1
+	$(INSTALLDIR) $(DESTDIR)$(PREFIX)/share/man/man5
+	$(INSTALLMAN) $(PROG).conf.5 $(DESTDIR)$(PREFIX)/share/man/man5/$(PROG).conf.5
 
 clean:
 	rm -f $(CLEANFILES)
