Index: debian_fdm/GNUmakefile
===================================================================
--- debian_fdm.orig/GNUmakefile	2008-04-27 08:37:20.000000000 +0200
+++ debian_fdm/GNUmakefile	2008-04-27 08:39:58.000000000 +0200
@@ -50,10 +50,13 @@
 DEFS+= -DNO_STRTONUM -DNO_SETRESUID -DNO_SETRESGID -DNO_SETPROCTITLE
 endif
 
-ifeq ($(shell uname),Linux)
+ifneq (, $(filter Linux GNU GNU/%, $(shell uname -s)))
 INCDIRS+= -I/usr/include/openssl -Icompat
 SRCS+= compat/strlcpy.c compat/strlcat.c compat/strtonum.c
-DEFS+= $(shell getconf LFS_CFLAGS) -D_GNU_SOURCE -DWITH_MREMAP \
+ifeq (, $(filter GNU/k%BSD, $(shell uname -s)))
+DEFS+= -DWITH_MREMAP
+endif
+DEFS+= $(shell getconf LFS_CFLAGS) -D_GNU_SOURCE \
        -DNO_STRLCPY -DNO_STRLCAT -DNO_STRTONUM -DNO_SETPROCTITLE \
        -DNO_QUEUE_H -DNO_TREE_H
 LIBS+= -lresolv
