Include a patch from Louis Opter to fix #538194.

Index: fdm-1.6/deliver-maildir.c
===================================================================
--- fdm-1.6.orig/deliver-maildir.c	2009-11-10 21:23:19.000000000 +0000
+++ fdm-1.6/deliver-maildir.c	2009-11-10 21:25:01.000000000 +0000
@@ -190,7 +190,7 @@
 	log_debug2("%s: writing to %s", a->name, src);
 	n = write(fd, m->data, m->size);
 	if (n < 0 || (size_t) n != m->size || fsync(fd) != 0)
-		goto error_unlink;
+		goto error_cleanup;
 	close(fd);
 	fd = -1;
 
@@ -199,24 +199,18 @@
 	 * back to find another name in the tmp directory.
 	 */
 	if (ppath(dst, sizeof dst, "%s/new/%s", path, name) != 0)
-		goto error_unlink;
+		goto error_cleanup;
 	log_debug2(
-	    "%s: linking .../tmp/%s to .../new/%s", a->name, name, name);
-	if (link(src, dst) != 0) {
+	    "%s: moving .../tmp/%s to .../new/%s", a->name, name, name);
+	if (safemove(src, dst) != 0) {
 		if (errno == EEXIST) {
-			log_debug2("%s: %s: link failed", a->name, src);
-			if (unlink(src) != 0)
-				fatal("unlink failed");
+			log_debug2("%s: %s: moving failed", a->name, src);
 			cleanup_deregister(src);
 			goto restart;
 		}
-		goto error_unlink;
+		goto error_cleanup;
 	}
 
-	/* Unlink the original tmp file. */
-	log_debug2("%s: unlinking .../tmp/%s", a->name, name);
-	if (unlink(src) != 0)
-		goto error_unlink;
 	cleanup_deregister(src);
 
 	/* Save the mail file as a tag. */
@@ -226,9 +220,7 @@
 	xfree(path);
 	return (DELIVER_SUCCESS);
 
-error_unlink:
-	if (unlink(src) != 0)
-		fatal("unlink failed");
+error_cleanup:
 	cleanup_deregister(src);
 
 error_log:
Index: fdm-1.6/fdm.h
===================================================================
--- fdm-1.6.orig/fdm.h	2009-11-10 21:25:01.000000000 +0000
+++ fdm-1.6/fdm.h	2009-11-10 21:25:01.000000000 +0000
@@ -865,6 +865,7 @@
 const char 	*checkmode(struct stat *, mode_t);
 const char 	*checkowner(struct stat *, uid_t);
 const char 	*checkgroup(struct stat *, gid_t);
+int		 safemove(const char *, const char *);
 
 /* mail.c */
 int		 mail_open(struct mail *, size_t);
Index: fdm-1.6/file.c
===================================================================
--- fdm-1.6.orig/file.c	2009-11-10 21:23:19.000000000 +0000
+++ fdm-1.6/file.c	2009-11-10 21:25:01.000000000 +0000
@@ -314,3 +314,39 @@
 	    "bad group: %lu, should be %lu", (u_long) sb->st_gid, (u_long) gid);
 	return (msg);
 }
+
+/*
+ * Move file oldpath to newpath. oldpath is always removed even in case of
+ * failure.
+ *
+ * It returns 0 on success and -1 on failure with errno set.
+ *
+ * This function use link + unlink or stat + rename if link fail with EXDEV.
+ * (see http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=538194)
+ */
+int
+safemove(const char *oldpath, const char *newpath)
+{
+	int		ret;
+	int		errsave;
+	struct stat	sb;
+
+	ret = link(oldpath, newpath);
+	if (ret != 0 && errno == EXDEV) {
+		ret = stat(newpath, &sb);
+		if (ret == -1) {
+			if (errno == ENOENT)
+				ret = rename(oldpath, newpath);
+		} else {
+			ret = -1;
+			errno = EEXIST;
+		}
+	}
+
+	errsave = errno;
+	if (unlink(oldpath) != 0 && errno != ENOENT)
+		fatal("unlink failed");
+	errno = errsave;
+
+	return (ret);
+}
